cmd/juju/application/removeapplication.go

```diff
@@ -241,6 +241,9 @@ func (c *removeApplicationCommand) removeApplications(
 		if result.Error != nil {
 			ctx.Infof("removing application %s failed: %s", name, result.Error)
 			anyFailed = true
+			if errors.IsNotSupported(err) {
+				ctx.Warningf("a change to application %s has been detected; retry shortly", name)
```

Comment:
This is cause stuttering as the warning will be shown as well as the error.
Ideally you'd simply detect not supported as you have done and replace the error with a new bespoke error relevant to the command, ie
```
err := result.Error
if errors.IsNotSupported(err) {
   err = errors.New("blah")
}
```